### PR TITLE
Add trace logging of PTP messages

### PR DESCRIPTION
**Pull request by jasonmartens**

### CODE REVIEW

File: src/ptp.rs [hunk 1]
```diff
@@ -764,7 +806,8 @@ impl<'a> PtpCamera<'a> {
 
             if mtype == PtpContainerType::Data && msg.is_response(&transaction) {
                 data = Some(msg.data);
-            } else if mtype == PtpContainerType::Response && msg.is_response(&transaction) {
+            }
```
Comment (tcr3dr) [inline src/ptp.rs]: weird

File: src/ptp.rs [hunk 2]
```diff
@@ -940,7 +984,8 @@ pub fn open_device(context: &mut libusb::Context, vid: u16, pid: u16) -> Option<
     None
 }
 
-pub fn find_readable_endpoint(device: &mut libusb::Device, device_desc: &libusb::DeviceDescriptor, direction: libusb::Direction, transfer_type: libusb::TransferType) -> Option<EndpointAddress> {
+pub fn find_readable_endpoint(device: &mut libusb::Device, device_desc: &libusb::DeviceDescriptor,
```
Comment (tcr3dr) [inline src/ptp.rs:987]: weird

File: src/ptp.rs [hunk 3]
```diff
@@ -110,9 +112,47 @@ pub trait PtpCommandCode: Sized + Copy {
...
         }
     }
+
+    fn enum_name(&self) -> String;
 }
 
-impl PtpCommandCode for StandardCommandCode {}
+impl PtpCommandCode for StandardCommandCode {
+    fn enum_name(&self) -> String {
+        match *self {
```
Comment (tcr3dr) [inline src/ptp.rs:122]: You should add a `use StandardCommandCode::*` here and move all the `StandardCommandCode::` prefixes below.

Also, does `format!("{:?}", self)` not produce the name of the enum? Just curious, might save hardcoding all this.

File: Cargo.toml [hunk 4]
```diff
@@ -12,4 +12,7 @@ libusb = "0.2.2"
 byteorder = "0.4.2"
 num = "0.1.30"
 enum_primitive = "0.1.0"
+env_logger = "0.3"
 encoding = "0.2.32"
+log = "0.3"
+time = "0.1"
```
Comment (tcr3dr) [inline Cargo.toml]: Please add newline at end of file

File: Cargo.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ptp"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (tcr3dr) [inline Cargo.toml]: Bump to 0.2.0 since we changed the public API of PtpCommandCode

File: src/lib.rs [hunk 6]
```diff
@@ -1,9 +1,13 @@
 #![allow(non_snake_case)]
 
 #[macro_use] extern crate enum_primitive;
+#[macro_use] extern crate log;
+
 extern crate libusb;
 extern crate byteorder;
 extern crate num;
 extern crate encoding;
+extern crate env_logger;
```
Comment (tcr3dr) [inline src/lib.rs]: env_logger doesn't look like it's used in this crate anywhere. Remove it?

File: src/ptp.rs (hunk 3 above)
Comment (jasonmartens) [inline src/ptp.rs:122]: Trying to add the use, but it's not working: 
src/ptp.rs:121:13: 121:32 error: unresolved import `StandardCommandCode::*`. Maybe a missing `extern crate StandardCommandCode`? [E0432]
src/ptp.rs:121         use StandardCommandCode::*;

Comment (jasonmartens) [inline src/ptp.rs:122]: core::fmt::Debug isn't implemented for Enum, which is why this is required. Plus, then Rust would have to allocate memory for all of the enumeration names, which seems crazy to do by default.

**Comment by tcr3dr** [comment]
lgtm

**Pull request merged**